packages/wp-source/src/libraries/handlers/author.ts

```diff
@@ -38,27 +38,34 @@ const authorHandler: Handler = async ({
     }
   });
 
-  // 3. populate response
+  // 3. Populate response.
   const items = await populate({ response, state });
   if (page > 1 && items.length === 0)
     throw new ServerError(`author "${slug}" doesn't have page ${page}`, 404);
 
-  // 4. get posts and pages count
-  const total = getTotal(response);
-  const totalPages = getTotalPages(response);
+  // 4. Get posts and pages count.
+  const total = getTotal(response, items.length);
+  const totalPages = getTotalPages(response, 0);
 
-  // 5. add data to source
+  // 5. Add data to source..
   const currentPageData = state.source.data[route];
   const firstPageData = state.source.data[path];
 
   Object.assign(currentPageData, {
+    link: route,
```

Comment:
@luisherranz The `link` property is redundant, right? But we still want to have it on the data object for each route...?

Reply:
I've found that when you pass down `data` like this:

```js
<Post data={data} />
```

it feels a bit weird to know everything about the post (the `type`, the `id`, the `isPost`...) but to get the `link` you have to go somewhere else to get it:

```js
const id = data.id;
const type = data.type;
const isPost = data.isPost;
const link = state.source[data.type][data.id].link // <-- feels overcomplicated.
```

or you have to pass it down along with `data`:

```js
<Post data={data} link={link} />
```

so I thought it made sense to store it in `data.link` along with the rest of the properties.

----

By the way maybe this type of discussion, not actually related to the implementation but to the design, could happen in the [feature topic](https://community.frontity.org/t/search-links-and-pagination-in-data-objects-of-wp-source/1230). What do you think?

Reply:
You're right, this is more of a discussion for the forum. I ll copy it there :) 

Just for reference, I agree with your reasoning here.